Src/Newtonsoft.Json/Utilities/BoxedPrimitives.cs

```diff
@@ -0,0 +1,83 @@
+﻿#region License
+// Copyright (c) 2022 James Newton-King
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+#endregion
+
+
+namespace Newtonsoft.Json.Utilities
+{
+    internal static class BoxedPrimitives
+    {
+        internal static object Get(bool value) => value ? BooleanTrue : BooleanFalse;
+        internal static readonly object BooleanTrue = true, BooleanFalse = false;
+
+        internal static object Get(int value)
+        {
+            switch (value)
+            {
+                case -1:
+                case 0:
+                case 1:
+                case 2:
+                case 3:
+                    return Int32Small[value + 1];
+                default:
+                    return value;
+            }
+        }
+        private static readonly object[] Int32Small = new object[] { -1, 0, 1, 2, 3 };
```

Comment:
How come there is a range of cached 32/64bit numbers, but only zero for decimal? Why up to 3?

Is there a reason to use an array here rather than separate fields? Fields would avoid array access.

Reply:
Fairly arbitrary based on observed use. Although in my "real" case, it was actually 0-5 that dominated. I can switch to fields, and: "pick a range" :)

Reply:
For decimal: as a gross over simplification - there don't *tend* to be as many common hits with decimals. Sure, we could pick a few integer values, but...

I'm open to suggestions